spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/configdata/KubernetesConfigDataLocationResolver.java

```diff
@@ -0,0 +1,130 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config.configdata;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+
+import org.springframework.boot.context.config.ConfigDataLocation;
+import org.springframework.boot.context.config.ConfigDataLocationNotFoundException;
+import org.springframework.boot.context.config.ConfigDataLocationResolver;
+import org.springframework.boot.context.config.ConfigDataLocationResolverContext;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.boot.context.config.Profiles;
+import org.springframework.boot.logging.DeferredLogFactory;
+import org.springframework.cloud.kubernetes.commons.KubernetesNamespaceProvider;
+import org.springframework.cloud.kubernetes.commons.config.ConfigMapConfigProperties;
+import org.springframework.cloud.kubernetes.commons.config.SecretsConfigProperties;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.StandardEnvironment;
+
+import static org.springframework.boot.cloud.CloudPlatform.KUBERNETES;
+import static org.springframework.util.ClassUtils.isPresent;
+
+/**
+ * @author Ryan Baxter
+ */
+public abstract class KubernetesConfigDataLocationResolver
+		implements ConfigDataLocationResolver<KubernetesConfigDataResource>, Ordered {
+
+	private static final boolean RETRY_IS_PRESENT = isPresent("org.springframework.retry.annotation.Retryable", null);
+
+	private final Log log;
+
+	public KubernetesConfigDataLocationResolver(DeferredLogFactory factory) {
+		this.log = factory.getLog(KubernetesConfigDataLocationResolver.class);
+	}
+
+	protected final String getPrefix() {
+		return "kubernetes:";
+	}
+
+	@Override
+	public final int getOrder() {
+		return -1;
+	}
+
+	@Override
+	public final boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDataLocation location) {
+		return location.hasPrefix(getPrefix())
+				&& (KUBERNETES.isEnforced(context.getBinder()) || KUBERNETES.isDetected(new StandardEnvironment()));
+	}
+
+	@Override
+	public final List<KubernetesConfigDataResource> resolve(ConfigDataLocationResolverContext context,
+			ConfigDataLocation location)
+			throws ConfigDataLocationNotFoundException, ConfigDataResourceNotFoundException {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public final List<KubernetesConfigDataResource> resolveProfileSpecific(
+			ConfigDataLocationResolverContext resolverContext, ConfigDataLocation location, Profiles profiles)
+			throws ConfigDataLocationNotFoundException {
+
+		ConfigDataProperties properties = ConfigDataProperties.of(resolverContext).register();
```

Comment:
the logic in this class stays exactly the same, but now it's a lot shorter and easier to read. `ConfigDataProperties.of(resolverContext).register()` this both computed all 3 properties needed (client, configmap and secret ones) + registers them.